src/App.jsx

```diff
@@ -14,33 +17,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const sortByABC = 'alphabetically';
+const sortByLength = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoods(goods, { sortField, reverseGood }) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case sortByABC:
+          return good1.localeCompare(good2);
+        case sortByLength:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reverseGood) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseGood, setReverse] = useState('');
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    reverseGood,
+  });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== sortByABC,
+          })}
+          onClick={() => {
+            setSortField(sortByABC);
+          }}
+        >
+          Sort alphabetically
+        </button>
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== sortByLength,
+          })}
+          onClick={() => {
+            setSortField(sortByLength);
+          }}
+        >
+          Sort by length
+        </button>
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !reverseGood,
+          })}
+          onClick={() => {
+            setReverse(!reverseGood);
+          }}
+        >
+          Reverse
+        </button>
+
+        {(sortField || reverseGood) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField('');
+              setReverse(false);
+            }}
+          >
+            Reset
+          </button>
+        )}
+      </div>
+
+      <ul>
+        {visibleGoods.map((good, index) => (
+          <li data-cy="Good" key={index}>
```

Comment:
```suggestion
          <li data-cy="Good" key={good}>
```
The key should be unique and stable